common/src/main/java/com/microsoft/identity/common/internal/util/QueryParamsAdapter.java

```diff
@@ -40,9 +44,14 @@
 
 /**
  * Class to serialize and deserialize query parameters from List<Pair<String, String>> to json String
- * and vice versa
+ * and vice versa.
+ *
+ * NOTE: Even we no longer use Pair (Since it's android-only), we are keeping this the same
+ *       to maintain backcompat with serialized value from older common that still uses it.
  */
-public class QueryParamsAdapter extends TypeAdapter<List<Map.Entry<String, String>>> {
+public class QueryParamsAdapter extends TypeAdapter<List<Pair<String, String>>> {
```

Comment:
How does this now work with new MSAL? I suppose that's going to send Map.Entry?

Reply:
We should document this in the release notes for MSAL/ADAL.. As part of how to upgrade to new major version.
right? 
<Changing the code to send Map.Entry instead of Pair>

Reply:
I think with this change now even new one sends Pair..so I don't expect there to be any changes needed for upgrade. @rpdome can confirm.

Reply:
the new one will serialize Map.Entry into the very same format as Pair.

Reply:
I had to make this change in Authenticator code, with the first RC build, that's why I thought..
Maybe I will need to revert that with the new RC build :)